indico/web/client/styles/widgets/_session-bar.scss

```diff
@@ -0,0 +1,35 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+@use 'base/palette' as *;
+
+#language-list {
```

Comment:
It's usually better to use classes for styling instead of IDs. And this may be a good case for scss nesting:

```scss
.session-bar-language-list {
  // ...

  button {
    //...
  }
}
```

Reply:
> It's usually better to use classes for styling instead of IDs. And this may be a good case for scss nesting:

I use `id` when it's a single element on the page as in this case. This makes it more obvious that this element is not going to be used in multiple places, and makes editing CSS safer. The idea that "classes are better" is quite an old opinion that has to do with inefficiency of the ID-based selector in the past, and today is no longer valid. In fact, much of the problems people claim is CSS's fault comes from the overuse of classes. I only use classes in two situiations:

- when I need to mark a group of elements that are *semantically* related and need to be differentiated from other elements
- when the element needs to be styled independently of its context (portable)

In this case, the language list does not satisfy the conditions.

As far as nesting goes, I tend to use explicit selectors when the selector is not intended to be `@extend`ed later.

Let me know if you would still insist to have these changed regardless.

Reply:
OK, fine for me. i'd still use nesting though ;)

Reply:
> OK, fine for me. i'd still use nesting though ;)

What's your preference regarding what nesting should represent? I generally avoid nesting to represent the relative relationships of elements because fixing that in a nested block is more work than fixing it in a combinator.

Reply:
I usually use it to avoid repetition. And in this case it nicely 'namespaces' the rules since they are all related to one parent component